### ISSUE TITLE
ErrorPresenter doesn't have access to GetRequestContext or GetResolverContext

### ISSUE THREAD

**Issue by vikstrous** [99designs/gqlgen#1067]
### What happened?
As of v0.11.0, when using NewDefaultServer, ErrorPresenter doesn't have access to GetRequestContext or GetResolverContext.

### What did you expect?
Old versions of gqlgen allowed the error presenter to inspect the graphql request. We used this to log errors with a lot of structured data about the request that they happened in. Some of the useful data we logged in errors:
- the resolver path that triggered the error
- the name(s) of the query

It seems like there's no easy way to get this information in the error presenter now. Is there another way to log this information when an error happens?

### Minimal graphql.schema and models to reproduce
n/a

### versions
 - `0.11.1` 
 - `1.13`
 - go modules

**Comment by vektah** [comment]
It should still work this way. The error presenter is now called more consistently for validation errors, which by definition dont have a valid request, nor request context.

Can you reproduce this?

**Comment by vikstrous** [comment]
Yes, it's deterministic. This is what the stack trace looks like, in case that helps.
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: missing operation context [recovered]
	panic: missing operation context
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1cee3e8]

goroutine 2486 [running]:
testing.tRunner.func1(0xc0014ddf00)
	/usr/local/go/src/testing/testing.go:874 +0x69f
panic(0x4eb2cc0, 0x5f4d7a0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/99designs/gqlgen/graphql.GetOperationContext(0x602b560, 0xc003fa9710, 0xe)
	/go/pkg/mod/github.com/99designs/gqlgen@v0.11.1/graphql/context_operation.go:58 +0xb9
redacted/gqlutil.errorPresenter(0x602b560, 0xc003fa9710, 0x5fc33a0, 0xc008542730, 0x5fc33a0)
	redacted/gqlutil.go:143 +0x79
github.com/99designs/gqlgen/graphql/handler.(*Server).ServeHTTP.func1(0xc000480b00, 0xc005483068, 0x601c5a0, 0xc00634de00)
	/go/pkg/mod/github.com/99designs/gqlgen@v0.11.1/graphql/handler/server.go:124 +0x251
panic(0x50025e0, 0x8128090)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
redacted/gqlutil.errorPresenter(0x602b560, 0xc003deea20, 0x5fc3aa0, 0xc0031e8000, 0xc001bca2c0)
	redacted/gqlutil.go:168 +0x408
github.com/99designs/gqlgen/graphql.AddError(0x602b560, 0xc003deea20, 0x5fc3aa0, 0xc0031e8000)
	/go/pkg/mod/github.com/99designs/gqlgen@v0.11.1/graphql/context_response.go:56 +0x140
github.com/99designs/gqlgen/graphql/handler.executor.DispatchError(0xc004003080, 0x57fed38, 0xc0040030a0, 0xc005325100, 0x1, 0x1, 0xc0053250f0, 0x1, 0x1, 0xc000480b00, ...)
	/go/pkg/mod/github.com/99designs/gqlgen@v0.11.1/graphql/handler/executor.go:171 +0x20e
ithub.com/99designs/gqlgen/graphql/handler/transport.POST.Do(0x601c5a0, 0xc00634de00, 0xc005741b00, 0x601c9e0, 0xc0085420a0)
/go/pkg/mod/github.com/99designs/gqlgen@v0.11.1/graphql/handler/transport/http_post.go:47 +0x58a
github.com/99designs/gqlgen/graphql/handler.(*Server).ServeHTTP(0xc000480b00, 0x601c5a0, 0xc00634de00, 0xc005741b00)
	/go/pkg/mod/github.com/99designs/gqlgen@v0.11.1/graphql/handler/server.go:140 +0x346
...
```

**Comment by vektah** [comment]
Ok, that doesnt look like a validation error. Ill take a closer look asap.

**Comment by vikstrous** [comment]
Any news on this?

**Comment by cliedeman** [comment]
Also managed to get this error. 
@vikstrous do you have any custom extensions enabled?

It definitely does not seem to happen during normal operation.

One case I can confirm is that if you send an invalid query I get the exception.

But I also saw an error that failed with this message:
```
input: operation  not found
```
https://github.com/99designs/gqlgen/blob/78f3da2296a5d69967a7ab09b66338ed9bd94033/graphql/executor/executor.go#L69

@vektah I think this can be fixed with an 
```
graphql.WithOperationContext(r.Context(), rc)
```
on line https://github.com/99designs/gqlgen/blob/aa407b1f3553ac2aee1939fbe28c85ed5cbfcdf9/graphql/handler/transport/http_post.go#L51
because the operation context is not inserted into the context in the failure case

**Comment by vikstrous** [comment]
I don't have any custom extensions. I have a very basic setup.

**Comment by cliedeman** [comment]
@vikstrous good to know. I was worried my custom extension was causing this issue

**Comment by gsgalloway** [comment]
Was encountering this error. Seems as though I might have been able to fix it by just replacing all calls to `graphql.GetRequestContext(ctx)` with `graphql.GetOperationContext(ctx)` and calls to `graphql.GetResolverContext(ctx)` with `graphql.GetFieldContext(ctx)`.

The deprecation warnings for these functions were helpful

**Comment by gsgalloway** [comment]
Was encountering this error. Seems as though I might have been able to fix it by just replacing all calls to graphql.GetRequestContext(ctx) with graphql.GetOperationContext(ctx) and calls to graphql.GetResolverContext(ctx) with graphql.GetFieldContext(ctx).

The deprecation warnings for these functions were helpful

**Comment by mtibben** [comment]
Fixed in #1154
